src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/extension.ts

```diff
@@ -182,6 +185,33 @@ export async function activate(vscodeType: typeof vscodeapi, context: ExtensionC
             localRegistrations.length = 0;
         });
 
+        const provider = new BlazorDebugConfigurationProvider(logger, vscodeType);
+        context.subscriptions.push(vscodeType.debug.registerDebugConfigurationProvider('blazor', provider));
+
+        /**
+         * On non-Windows platforms, we need to terminate the Blazor
+         * dev server and its child processes.
+         */
+        vscodeType.debug.onDidTerminateDebugSession(event => {
+            logger.logVerbose('Terminating debugging session...');
+            if (process.platform !== 'win32') {
+                psList().then(processes => {
+                    const devserver = processes.find(
+                        (process: psList.ProcessDescriptor) => !!(process && process.cmd && process.cmd.includes('blazor-devserver')),
+                    );
+                    if (devserver) {
+                        const command = `kill ${devserver.pid}`;
+                        exec(command, (error, stdout, stderr) => {
+                            if (error) {
+                                logger.logError('Error during debug process clean-up: ', error);
+                            }
+                            return logger.logMessage('Debug process clean-up complete.');
```

Comment:
Given the "Terminating debugging session..." is verbose should this also be? Also I'd imagine all of these logs should probably also have a prefix like in the provider.